Core/src/org/sleuthkit/autopsy/textextractors/TextFileExtractor.java

```diff
@@ -94,33 +102,59 @@ public boolean isSupported() {
         return file.getMIMEType().equals("text/plain");
     }
 
-    public static Charset getEncoding(Content content) {
-        try (InputStream stream = new BufferedInputStream(new ReadContentInputStream(content))) {
-            // Tika first
+    /**
+     * Return the encoding of the file
+     * @return Detected encoding or UNKNOWN_CHARSET 
+     */
+    public Charset getEncoding() {
+        if (encoding != null)
+            return encoding;
+        
+        // Encoding detection is hard. We use several libraries since the data passed in is often messy.
+        
+        // First try CharsetDetector (from Tika / ICU4J)
+        // It is a rule-baesd detection approach
+        try (InputStream stream = new BufferedInputStream(new ReadContentInputStream(file))) {
             CharsetDetector detector = new CharsetDetector();
             detector.setText(stream);
             CharsetMatch tikaResult = detector.detect();
-            if (tikaResult != null && tikaResult.getConfidence() >= MIN_TIKA_MATCH_CONFIDENCE) {
+            if (tikaResult != null && tikaResult.getConfidence() >= MIN_CHARSETDETECT_MATCH_CONFIDENCE) {
                 try {
-                    return Charset.forName(tikaResult.getName());
-                } catch (UnsupportedCharsetException ignored) {
+                    encoding = Charset.forName(tikaResult.getName());
+                    return encoding;
+                } catch (UnsupportedCharsetException ex) {
+                    logger.log(Level.WARNING, "Error converting CharsetDetector Result", ex);
```

Comment:
We should log the object ID of the file to give this message some context. I will integrate this PR and then circle back and augment the logging.